src/data_api/orders.se

```diff
@@ -155,105 +155,79 @@ def getWorstBidOrderID(market: address, outcome: uint256):
+        isWorstPrice = fxpPrice >= self.orders[worseOrderID].fxpPrice
+    if(isWorstPrice):
+        return([worseOrderID, self.orders[worseOrderID].worseOrderID]: arr)
+    isBetterPrice = self.isBetterPrice(type, fxpPrice, worseOrderID)
+    while(isBetterPrice and self.orders[worseOrderID].betterOrderID != 0 and fxpPrice != self.orders[self.orders[worseOrderID].betterOrderID].fxpPrice):
+        isBetterPrice = self.isBetterPrice(type, fxpPrice, self.orders[worseOrderID].betterOrderID)
+        if(isBetterPrice):
+            worseOrderID = self.orders[worseOrderID].betterOrderID
+    betterOrderID = self.orders[worseOrderID].betterOrderID
+    return([betterOrderID, worseOrderID]: arr)
 
-# @private
-def updateWorstAskOrder(orderID: address, market: address, fxpPrice, outcome: uint256, worseOrderID: uint256):
-    if(msg.sender != self):
-        ~invalid()
-    worstAskOrderID = self.worstOrder[market][outcome].ask
-    # If worst ask is not set, this order is the new worst ask
-    if(worstAskOrderID == 0):
-        self.worstOrder[market][outcome].ask = orderID
+# @public
+def descendOrderList(type: uint256, fxpPrice, highestOrderID: address):
+    betterOrderID = highestOrderID
+    if(type == BID):
+        isBestPrice = fxpPrice > self.orders[betterOrderID].fxpPrice
     else:
-        # If price higher than worst ask price, this order is the new worst ask
-        if(fxpPrice > self.orders[worstAskOrderID].fxpPrice):
-            if(worseOrderID != 0):
-                ~invalid()
-            self.worstOrder[market][outcome].ask = orderID
-        # If price is lower than the worst ask price:
-        #  - worseOrderID must be specified
-        #  - price must be less than or equal to worseOrderID's price
-        else:
-            if(worseOrderID == 0):
-                ~invalid()
-            if(fxpPrice > self.orders[worseOrderID].fxpPrice):
-                ~invalid()
-    return(1: uint256)
+        isBestPrice = fxpPrice < self.orders[betterOrderID].fxpPrice
+    if(isBestPrice):
+        return([0, betterOrderID]: arr)
+    if(fxpPrice == self.orders[betterOrderID].fxpPrice):
+        return([betterOrderID, self.orders[betterOrderID].worseOrderID]: arr)
+    isWorsePrice = self.isWorsePrice(type, fxpPrice, betterOrderID)
+    while(isWorsePrice and self.orders[betterOrderID].worseOrderID != 0):
+        isWorsePrice = self.isWorsePrice(type, fxpPrice, self.orders[betterOrderID].worseOrderID)
+        if(isWorsePrice or fxpPrice == self.orders[self.orders[betterOrderID].worseOrderID].fxpPrice):
+            betterOrderID = self.orders[betterOrderID].worseOrderID
+    worseOrderID = self.orders[betterOrderID].worseOrderID
+    return([betterOrderID, worseOrderID]: arr)
+
+# @public
+def findBoundingOrders(type: uint256, fxpPrice, bestOrderID: address, worstOrderID: address, betterOrderID: address, worseOrderID: address):
+    if(bestOrderID == worstOrderID):
+        return([bestOrderID, bestOrderID]: arr)
```

Comment:
This is done in the next step -- the new best/worstOrderIDs are looked up, then compared to the current order.  It could be worth re-doing this comparison in `findBoundingOrders` to make its output easier to understand (if it's considered as a standalone function).